EIPS/eip-7825.md

```diff
@@ -0,0 +1,74 @@
+---
+eip: 7825
+title: Transaction Gas Limit Cap
+description: Introduce a protocol-level cap on the maximum gas used by a transaction to 30 million.
+author: Giulio Rebuffo (@Giulio2002)
+discussions-to: https://ethereum-magicians.org/t/eip-7825-transaction-gas-limit-cap/21848
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-11-23
+---
+
+## Abstract
+
+This proposal introduces a protocol-level cap on the maximum gas usage per transaction to 30 million gas. By implementing this limit, Ethereum can enhance its resilience against certain DoS vectors, improve network stability, and provide more predictability to transaction processing costs, especially in the context of increasing the Gas Limit.
+
+## Motivation
+
+Currently, transactions can theoretically consume up to the entire block gas limit, which poses several risks:
+
+1. **DoS Attacks**: A single transaction consuming most or all of the block gas can result in uneven load distribution and impact network stability.  
+2. **State Bloat Risks**: High-gas transactions often result in larger state changes, increasing the burden on nodes and exacerbating the Ethereum state growth problem.  
+3. **Validation Overhead**: High-gas transactions can lead to longer block verification times, negatively impacting user experience and network decentralization.
+
+By limiting individual transactions to a maximum of 30 million gas, we aim to:
+
+- Reduce the risks of single-transaction DoS attacks.  
+- Promote fairer gas allocation across transactions within a block.  
+- Ensure better synchronization among nodes by mitigating extreme block validation times.
+
+## Specification
+
+### Gas Cap
+
+- Enforce a protocol-level maximum of **30 million gas** for any single transaction.  
+- This cap applies regardless of the block gas limit set by miners or validators.  
+- Transactions specifying gas limits higher than 30 million gas will be rejected with an appropriate error code (e.g., `MAX_GAS_LIMIT_EXCEEDED`).  
+
+### Changes to EVM Behavior
+
+1. **Validation**: During transaction validation, if the `gasLimit` specified by the sender exceeds 30 million, the transaction is invalidated (not included in the txpool). 
+2. **Execution**: At the start of execution, if the transaction exceeds the cap, execution halts immediately, and the sender is refunded for unspent gas, and the block is deemed invalid.
```

Comment:
```suggestion
2. **Block Validation**: As part of block validation before processing, any block having a transaction with `gasLimit` > 30 million is deemed invalid and rejected.
```

we don't need execution case because that is OOG case when the execution exceeds its gasLimit